tasks/config/run.js

```diff
@@ -62,77 +62,88 @@ module.exports = function (grunt) {
     '--server.port=5610',
   ];
 
+  const githubChecksReporter = title => ['run', 'github-checks-reporter', title, 'node'];
+  const YARN = 'yarn';
```

Comment:
I think we should clean this up a bit by defining each task config with a helper, rather than splitting up the responsibilities into such tiny chunks.

I also think we should avoid using the grunt config identifiers. I'd prefer if the names of each of these checks represented the script that used to execute them, maybe by using the script filename as the check name, like `scripts/eslint` and `scripts/check_file_casing`.

I assume something like `scripts/functional_tests --config test/api_integration/config.js` doesn't fit well in the Github UI, so maybe we should create `scripts/test_api_integration`, `scripts/test_server_integration`, etc. which basically just set default arguments for `scripts/functional_tests`, and then use them... We might want to set this up as a separate effort though.

```js
eslint: scriptWithGithubChecks({
  name: 'eslint',
  args: ['--no-cache']
}),

apiIntegrationTests: scriptWithGithubChecks({
  name: 'scripts/functional_tests',
  title: 'API Integration Tests',
  args: [
    '--config', 'test/api_integration/config.js',
    '--bail',
    '--debug'
  ]
})
```

```js
const scriptWithGithubChecks = ({ name, title = name, args = [] }) => ({
  cmd: 'yarn',
  args: [
    'run',
    'github-checks-reporter',
    `scripts/${name}`,
    'node',
    `scripts/${name}`
    ...args
  ]
})
```